winch/codegen/src/isa/x64/asm.rs

```diff
@@ -32,6 +32,47 @@ pub(crate) enum Operand {
     Imm(i64),
 }
 
+/// x64 supported comparisons.
+pub(crate) enum CmpKind {
```

Comment:
Opting to do this for two reasons. (1) I think exposing the `CC` enum at this level mixes our layers of abstractions because it's tied to the `setcc` instruction and not `cmp`. (2) `CC` includes variants that don't make sense in a `cmp` context so allowing them to be passed in from the macroassembler seems like it could potentially lead to unwanted behaviour.

Reply:
> Is there a reason why we can't go from the `CmpKind` defined at the masm layer directly to CC?

No strong reason, I'll make that change. I had thought it would add a circular dependency but thinking more about it, it doesn't.